### DISCUSSION TITLE
Connecting RGB LED to I2C?

### DISCUSSION THREAD

**Discussion by redm123** [arendst/Tasmota#22100] [Q&A]
Hi all! I looking for some idea how to do the following:

I have BME680 sensor on a little breakout board connected via I2C. That sensor lives in a small case, connected with a few meter 4-wire cable (to a Wemos). 
Now I'd like to extend that by some status single RGB LED (attached to the case). The LED does not need to do anything fancy, but display some static colors, and adjustable brightness. To make it easier to mount, the LED should be "external" (not glued to a BOB)
The only option I see is also connecting that to the I2C. But I couldn't find any suitable I2C LED controller on some small enough board. There are Sparkfun LP55231 or Adafruit AW9523 which could probably do the job. However with 4-5cm they are too big.
Oh, and it should of course be controllable by Tasmota. I think I could tweak some I2C driver, if needed. It's the hardware side that leave me a bit clueless...

Any suggestions, ideas?

**Comment by HofeBY** [comment]
without I2C:
WS2812 is what you search, RGB with brightness and several WS2812 on only a 1Wire Bus.

**Reply by redm123** [reply]
Yea... just that I2C is kind of the requirement, I think. Cause that's what I have in that little box with the sensor...
Or can I "tunnel" WS2812 over I2C?

**Reply by HofeBY** [reply]
No, WS2812 over I2C is not possible, but it´s only one more wire to your I2C Bus sharing GND & VCC.
But you can use PCF9685 16xPWM with three LED´s in red green blue. Have fun in scripting.
You can also put a small ESP including WS2812&programming Unit up to BME680 using only 1 USB cable.

https://de.aliexpress.com/item/1005006851575663.html?pm=a2g0o.order_list.order_list_main.5.7b785c5f1VQl7o&gatewayAdapt=glo2deu

Well, it shouldn't be that difficult to use a housing that is well ventilated on both sides and to install a thermally insulating sponge between the sensor and the ESP.

HofeBY

**Reply by sfromis** [reply]
An I/O expander on the I2C bus would also be an option, albeit without the variable brightness with PWM. But do note that I2C is designed for relatively short wiring, and "a few meter" is **way** outside of what can be expected to work ok.

But yeah, a small ESP32-C3 board in the case with the BME680 would also be a great option, with only power over your 4-wire cable.

**Reply by redm123** [reply]
Thanks for the input. An extra small ESP, I hadn't thought about...

I also now found an "Adafruit NeoDriver I2C", together with a single Through-Hole NeoPixel LED, that also sounds like it could work (if I remove that bulky connector).

> But do note that I2C is designed for relatively short wiring, and "a few meter" is way outside of what can be expected to work ok.

Hmm, I never thought about what it was designed for 😅 I had my constraints from reality (power socket, useful place for sensors, useful place for main ESP) and just tried it. And at least for the BME680, over 10m probably, with all the corners, it works reliably for a few years now 🤞

**Reply by sfromis** [reply]
While that Adafruit NeoDriver board may fit your plan, it is a bit bigger than a small ESP32-C3 board, and without a Tasmota driver already existing for it, you'd also have to create this.

Here's an example of what I mean, picked due to already having neopixels on the board. A lot more than a single one, but a 5x5 matrix to have fun with.
https://github.com/01Space/ESP32-C3FH4-RGB
Of course, this also works with the neopixel support already part of Tasmota, even including animations if your are in the mood for such.

However, beware that having the ESP32-C3 board close to the BME680 in a small case would be a bad plan, due to the heat from the MCU. Likewise, even adding a single neopixel with that Adafruit board would be problematic, as electronics has a bad habit of generating heat, leading to temperature and humidity readings becoming off, not correctable with an offset.

**Reply by redm123** [reply]
Yes, the heat problem is something I'm also a bit concerned about. I mean, that's one reason why I didn't put the main ESP in that case in the first place. And then maybe that NeoDriver + single LED is still better than the ESP32-C3... (Although this also sounds interesting). ...but it would be neat to have the LED in that case :)
I hope that I may get away with some good placement of the parts and maybe some internal shielding... Also I may only light the LED when there is status to report, and not all the time. But this remains to be evaluated in practice.

> without a Tasmota driver already existing for it, you'd also have to create this.

Yes, that's a bit a downside. We'll see, I'm still evaluating options... Thanks for the input anyway!
